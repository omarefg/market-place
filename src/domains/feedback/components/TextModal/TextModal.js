import React from "react";
import { createPortal } from "react-dom";
import classes from "./TextModal.module.css";

export function TextModal(props) {
  const { open, text, portal, onClose } = props;

  if (open) {
    return createPortal(
      <div
        className={classes.portal}
        onClick={onClose}
        data-testid="TextModal_container"
      >
        <p data-testid="TextModal_content">{text}</p>
      </div>,
      portal
    );
  }
  return null;
}
