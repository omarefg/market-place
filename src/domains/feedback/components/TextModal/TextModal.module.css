.portal {
  min-height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.9);
  z-index: 10;
  color: var(--color-3);
}
