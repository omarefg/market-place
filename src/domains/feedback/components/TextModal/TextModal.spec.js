import React from "react";
import { fireEvent, render, within } from "@testing-library/react";
import { TextModal } from "./TextModal";

const defaultProps = {
  open: false,
  text: "TextMock",
  portal: document.createElement("div"),
  onClose: jest.fn(),
};

const wrapper = (props = {}) => {
  const finalProps = { ...defaultProps, ...props };
  return render(<TextModal {...finalProps} />);
};

describe("@domains/feedback/components/TextModal", () => {
  it("Given a valid call it must render without crashing", () => {
    // Arrange
    const { container } = wrapper();
    // Assert
    expect(container).toMatchSnapshot();
  });

  it("Given a true `open` prop it must render the modal inside the portal", () => {
    //Arrange
    wrapper({ open: true });
    const { getByTestId } = within(defaultProps.portal);
    const content = getByTestId("TextModal_container");
    // Act
    const isChild = content.parentNode.isEqualNode(defaultProps.portal);
    // Assert
    expect(isChild).toBe(true);
  });

  it("Given a true `open` prop and a valid `text` prop it must render the passed text inside the portal", () => {
    //Arrange
    wrapper({ open: true });
    const { getByTestId } = within(defaultProps.portal);
    const { innerHTML } = getByTestId("TextModal_content");
    // Assert
    expect(innerHTML).toBe(defaultProps.text);
  });

  it("Given a true `open` prop and a valid `onClose` prop it must call the onClose prop when content is clicked", () => {
    // Arrange
    wrapper({ open: true });
    const { getByTestId } = within(defaultProps.portal);
    const content = getByTestId("TextModal_container");
    //Act
    fireEvent.click(content);
    // Assert
    expect(defaultProps.onClose).toBeCalledTimes(1);
  });
});
