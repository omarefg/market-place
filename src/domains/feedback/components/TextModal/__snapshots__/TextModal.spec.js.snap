// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@domains/feedback/components/TextModal Given a valid call it must render without crashing 1`] = `<div />`;
